test/containers_test.ml

```diff
@@ -1,6 +1,101 @@
+open Containers
+open Test
 
+(* Dummy tests for test runners *)
 
-(* Dummy tests for the unit test interface *)
-let () = Test.test (fun _ -> true)
+let () = test (fun t ->
+    truthy t true;
+    falsy t false
+)
 
-let () = Test.test_with_msg "dummy test" (fun x -> Test.truthy x true)
\ No newline at end of file
+(* Int64 module tests *)
+
+(* Intentionally not tested as Int64 module is just a wrapper around stdlib, this should be part of BS testing *)
+
+
+(* Map module tests *)
+module IntMap = Map.Make(Int64)
+
+let to_int64 i = match Int64.of_int i  with
+        | Some x -> x
+        | None -> raise (Failure "Corrupted implementation of Int64.of_int")
+
+let map0 = IntMap.add (to_int64 1) "one" IntMap.empty
+
+let () = test_with_msg "map.size" (fun t -> 
+    let x = IntMap.size IntMap.empty in
+    truthy t (x = 0)
+)
+
+let () = test_with_msg "map.size" (fun t -> 
+    let x = IntMap.size map0 in
+    truthy t (x = 1)
+)
+
+let () = test_with_msg "map.get" (fun t ->
+    let x = IntMap.get (to_int64 1) map0 in match x with
+    | Some str -> truthy t (str = "one")
+    | None -> truthy t false
```

Comment:
These could be greatly simplified by providing an `eq` function to compare values directly. E.g.
```ml
eq (IntMap.get (to_int64 1) map0) (Some "one")
```